packages/libs/client-utils/src/built-in/create-principal.ts

```diff
@@ -0,0 +1,27 @@
+import { addData, execution } from '@kadena/client/fp';
+import { pipe } from 'ramda';
+import { dirtyReadClient } from '../core/client-helpers';
+import type { IClientConfig } from '../core/utils/helpers';
+
+interface ICreatePrincipalInput {
+  keyset: {
+    keys: string[];
+    pred?: 'keys-all' | 'keys-2' | 'keys-any';
+  };
+}
+
+/**
+ * @alpha
+ */
+export const createPrincipal = async (
```

Comment:
It would be nice if more than just  principals based on keysets could be created. Mainly `capability-guards` should probably be supported.

```
(create-principal (create-capability-guard (some.module.CAPABILITY_NAME)))
```

Reply:
as discussed, saving this for another PR.